### PR TITLE
(DO NOT MERGE)(maint) Minor file encoding changes

### PR DESCRIPTION
**Pull request by Iristyle**
These are leftovers that remain after #5545 and #5384 have been merged.

They probably should end up in separate tickets, but put up to gather additional feedback.

### CODE REVIEW

**Comment by adrienthebo** [comment]
@Iristyle you have a `(WIP)` in one of the commits, so the commit format checker is mad at you.

**Comment by Iristyle** [comment]
#5593 / #5594 have spawned out of this and moved to their own tickets.

File: spec/unit/util/json_lockfile_spec.rb [hunk 1]
```diff
@@ -10,14 +10,30 @@
...
+  # 4-byte 𠜎 - http://www.fileformat.info/info/unicode/char/2070E/index.htm - 0xF0 0xA0 0x9C 0x8E / 240 160 156 142
+  let (:mixed_utf8) { "A\u06FF\u16A0\u{2070E}" } # Aۿᚠ𠜎
+
   describe "#lock" do
     it "should create a lock file containing a json hash" do
-      data = { "foo" => "foofoo", "bar" => "barbar" }
+      data = { "foo" => "foofoo", "bar" => "barbar", mixed_utf8 => mixed_utf8 }
       @lock.lock(data)
 
+      # TODO: check into the string encoding details here
```
Comment (Iristyle) [inline spec/unit/util/json_lockfile_spec.rb]: TODO

File: spec/unit/util/lockfile_spec.rb [hunk 2]
```diff
@@ -115,4 +115,17 @@ def self.run_in_forks(count, &blk)
...
+  it "should return UTF-8 lock data" do
+    # different UTF-8 widths
+    # 1-byte A
+    # 2-byte ۿ - http://www.fileformat.info/info/unicode/char/06ff/index.htm - 0xDB 0xBF / 219 191
+    # 3-byte ᚠ - http://www.fileformat.info/info/unicode/char/16A0/index.htm - 0xE1 0x9A 0xA0 / 225 154 160
+    # 4-byte ܎ - http://www.fileformat.info/info/unicode/char/2070E/index.htm - 0xF0 0xA0 0x9C 0x8E / 240 160 156 142
+    mixed_utf8 = "A\u06FF\u16A0\u{2070E}"  # Aۿᚠ܎
+
+    @lock.lock(mixed_utf8)
+    # TODO: bytes match, but the strings encodings are now different
```
Comment (Iristyle) [inline spec/unit/util/lockfile_spec.rb]: TODO

**Comment by adrienthebo** [comment]
@Iristyle we're getting a spec failure on Ruby 1.9.3 (https://travis-ci.org/puppetlabs/puppet/jobs/200548499#L231), care to take a gander at that?

**Comment by melissa** [comment]
https://tickets.puppetlabs.com/browse/PUP-9261 opened to finish up this work and get it in as a part of the agent locking epic

**Pull request closed without merging**